Hearthstone Deck Tracker/Hearthstone/GameV2.cs

```diff
@@ -31,6 +31,12 @@ public class GameV2 : IGame
 		private Mode _currentMode;
 		private BrawlInfo _brawlInfo;
 		private BattlegroundRatingInfo _battlegroundsRatingInfo;
+		const string UntransformedArannaCardid = HearthDb.CardIds.NonCollectible.Neutral.ArannaStarseekerTavernBrawl1;
+		const string TransformedArannaCardid = HearthDb.CardIds.NonCollectible.Neutral.ArannaStarseeker_ArannaUnleashedTokenTavernBrawl;
+		private Dictionary<string, string> _lastKnownBoardStateLookup = new Dictionary<string, string>()
+		{
+			{ TransformedArannaCardid, UntransformedArannaCardid}
+		};
```

Comment:
Instead of adding more things go GameV2 here, let's make a new `BattlegroundsBoardState` class that contains all this and the `LastKnownBattlegroundsBoardState` dictionary (which can be renamed to something simpler then). 

The class should probably look something like this
```csharp
public void UpdateSnapshot(string opponentHeroCardId); // looks up the key and sets it in the dict

public BoardSnapshot GetSnapshot(string  opponentHeroCardId); // looks up the key and returns the result from the dict
```

This will make it completely opaque to the implementation (GameV2) that there is any card id manipulation going on, and nothing but that class will need to care how the snapshots are keyed.